fluss-client/src/main/java/com/alibaba/fluss/client/table/FlussTable.java

```diff
@@ -207,17 +229,20 @@ public CompletableFuture<LookupResult> lookup(InternalRow key) {
 
     @Override
     public CompletableFuture<PrefixLookupResult> prefixLookup(InternalRow bucketKey) {
-        if (!hasPrimaryKey) {
-            throw new FlussRuntimeException(
-                    String.format("None-pk table %s don't support prefix lookup", tablePath));
-        }
-        // TODO: add checks the bucket key is prefix of primary key
+        maybeInitPrefixLookup();
```

Comment:
Bucket key shouldn't contain partition keys, but we need to extract the partition key from the prefix key. So current interface can't satisfy it, we need to rafctor the interface like `getLogScanner`:

```java
interface Table {
  PrefixLookuper getPrefixLookuper(PrefixLookup);
}

class PrefixLookup {
   // 1. lookupColumnNames should contain partition fields 
   // 2. (lookupColumnNames exclude partition fields) should be a prefix of (primary key exclude partition fields)
   PrefixLookup(String[] lookupColumnNames)
}

interface PrefixLookuper {
    CompletableFuture<PrefixLookupResult> prefixLookup(InternalRow prefixKey);
}
```

Reply:
Done!